Sources/Features/Breadcrumb/BacktraceComponentListener.swift

```diff
@@ -0,0 +1,126 @@
+import Foundation
+
+@objc class BacktraceComponentListener: NSObject {
+ 
+    override init() {
+        super.init()
+        observeOrientationChange()
+        observeMemoryStatusChanged()
+        observeBatteryStatusChanged()
+    }
+
+    deinit {
+        NotificationCenter.default.removeObserver(self, name: UIDevice.orientationDidChangeNotification, object: nil)
+        self.source?.cancel()
+    }
+    
+//MARK: Orientation Status Listener
+    
+    private func observeOrientationChange() {
+        NotificationCenter.default.addObserver(self, selector: #selector(notifyOrientationChange), name: UIDevice.orientationDidChangeNotification, object: nil)
+    }
+    
+    @objc private func notifyOrientationChange() {
+        switch UIDevice.current.orientation{
+            case .portrait, .portraitUpsideDown:
+                addBreadcrumb("portrait")
+            case .landscapeLeft, .landscapeRight:
+                addBreadcrumb("landscape")
+            default:
+                print("unknown")
+        }
+    }
+    
+    private func addBreadcrumb(_ orientation: String) {
```

Comment:
the name of the method gives you a generic information about the purpose of the method. I disagree - the addBreadcrumb method adds breadcrumb with the orientation setting